src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -156389,5 +156365,239 @@ public class C
+        (default(A<string?>)).F.ToString();
+        (default(A<string?>?)).F.ToString();
+    }
+}
+""");
+            comp.VerifyDiagnostics(
+                // (5,16): warning CS8618: Non-nullable field 'F' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.
+                //     internal T F;
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "F").WithArguments("field", "F").WithLocation(5, 16),
+                // (12,9): warning CS8602: Dereference of a possibly null reference.
+                //         (default(A<string?>)).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "(default(A<string?>)).F").WithLocation(12, 9),
+                // (12,10): warning CS8602: Dereference of a possibly null reference.
+                //         (default(A<string?>)).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(A<string?>)").WithLocation(12, 10),
+                // (13,10): warning CS8602: Dereference of a possibly null reference.
+                //         (default(A<string?>?)).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "default(A<string?>?)").WithLocation(13, 10));
+        }
+
+        [Fact, WorkItem(63425, "https://github.com/dotnet/roslyn/issues/63425")]
+        public void DefaultOfT4()
+        {
+            var comp = CreateCompilation(new[] { """
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+class C<T>
+{
+    void M1()
+    {
+        if (default(T) is not null)
+        {
+            M2(default);
+            M3(default);
+        }
+    }
+
+    void M2()
+    {
+        M2(default); // 1
+        M3(default);
+    }
+
+    void M3()
+    {
+        M3(default); // 2
+        M2(default);
+    }
+
+    void M2(T t) { }
+    void M3([DisallowNull] T t) { }
+}
+""", DisallowNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (18,12): warning CS8604: Possible null reference argument for parameter 't' in 'void C<T>.M2(T t)'.
+                //         M2(default); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "default").WithArguments("t", "void C<T>.M2(T t)").WithLocation(18, 12),
+                // (24,12): warning CS8604: Possible null reference argument for parameter 't' in 'void C<T>.M3(T t)'.
+                //         M3(default); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "default").WithArguments("t", "void C<T>.M3(T t)").WithLocation(24, 12));
+        }
+
+        [Fact, WorkItem(63425, "https://github.com/dotnet/roslyn/issues/63425")]
+        public void DefaultOfT5()
+        {
+            var comp = CreateCompilation(new[] { """
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+class C<T>
+{
+    void AssertNotNull([NotNull] T t) { }
+
+    void M1()
+    {
+        AssertNotNull(default);
+        default(T).ToString(); // ok
```

Comment:
This currently produces a wrong warning. Is it okay to skip the test and open a follow-up issue?

Reply:
Instead of skipping the test, let's record what it's currently doing (i.e. pass the value to VerifyDiagnostics that makes the test pass), then add a comment indicating what the correct behavior would be.

Basically, if the product somehow changes to make this scenario start working differently, we should know about it automatically by this test starting to fail.